### PR TITLE
Adding Functional Job Requirements and GH Actions

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,7 +34,19 @@ metadata:
...
+🚀	Though deployment and validation of your solution on the target deployment option is helpful, it does not provide the required level of quality assurance for functional verification, which is generally achieved with a test framework and automation normally included in the CI/CD cycle of the Partner product.
+
+## Functional Job Requirements
+
+1. Functional test should base its test cases on the specifications of the ISV product under test
+2. Functional test should validate the functionality of the ISV product and describe what the ISV product does
+3. Healthchecks, service endpoints checks or any other technical checks are not considered and accepted as valid functional test
+4. Functional test should be wrapped as a container and submitted as a Kubernetes job
```
Comment (shapirov103) [inline README.md:44]: Let's add here, container image should be published on ECR. I am concerned they will put something with vulnerability. 
Or we can say that in the future such container images will be required to publish on ECR and/or provide evidence of successful recent vulnerability scan.

File: README.md [hunk 2]
```diff
@@ -34,7 +34,19 @@ metadata:
...
+5. Functional test should enable broader test coverage and better customer experience
```
Comment (shapirov103) [inline README.md:45]: This seems like a fluff - it is very subject what "broader" and "better" means. We need unambiguous rules. 
So I suggest replace it with "should not require elevated security permissions, such as cluster roles, privileged mode, non-ephemeral storage"  - the storage one may be relaxed in the future if there is a clear evidence of the need.

File: README.md [hunk 3]
```diff
@@ -34,7 +34,19 @@ metadata:
...
+🚀	Though deployment and validation of your solution on the target deployment option is helpful, it does not provide the required level of quality assurance for functional verification, which is generally achieved with a test framework and automation normally included in the CI/CD cycle of the Partner product.
+
+## Functional Job Requirements
+
+1. Functional test should base its test cases on the specifications of the ISV product under test
```
Comment (shapirov103) [inline README.md:41]: Let's add somewhere here that 
1. Functional test must be implemented as a Kubernetes Job and any non-zero exit status of the job execution will be considered a failure.
2. Functional test must be repeatable. That means that if the job has executed before successfully and no changes were applied, we expect to run it continuously and mark the product as failure if the test job starts producing failures even if previous executions against the same environment were successful.

**Review by shapirov103 — COMMENTED** [review]
Looks good, a few comments, questions.

File: README.md [hunk 4]
```diff
@@ -34,7 +34,19 @@ metadata:
...
+2. Functional test should validate the functionality of the ISV product and describe what the ISV product does
+3. Healthchecks, service endpoints checks or any other technical checks are not considered and accepted as valid functional test
```
[the remark this answers is not in the document]
Comment (elamaran11) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (elamaran11) [reply README.md]: done.

File: README.md [hunk 5]
```diff
@@ -34,7 +34,21 @@ metadata:
...
+🚀	Though deployment and validation of your solution on the target deployment option is helpful, it does not provide the required level of quality assurance for functional verification, which is generally achieved with a test framework and automation normally included in the CI/CD cycle of the Partner product.
+
+## Functional Job Requirements
+
+1. Functional test should base its test cases on the specifications of the ISV product under test
+2. Functional test should validate the functionality of the ISV product and describe what the ISV product does
+3. Healthchecks, service endpoints checks or any other technical checks are do not represent sufficient coverage required for the functional test
```
Comment (shapirov103) [inline README.md:43]: "are do not" ->  do not

File: README.md [hunk 6]
```diff
@@ -34,7 +34,21 @@ metadata:
...
+4. Functional test should be wrapped as a container, container image should be published to ECR, and/or provide evidence of successful recent vulnerability scan
```
Comment (shapirov103) [inline README.md:44]: published on ECR

**Review by shapirov103 — COMMENTED** [review]
looks good, minor typos

File: README.md (hunk 5 above)
Comment (elamaran11) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (elamaran11) [reply README.md]: done

**Comment by elamaran11** [comment]
> looks good, minor typos

Fixed, Please merge this.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**